skycoin/skywire
# Fix/app state

Did you run `make format && make check`?
yes

Fixes #1209 
Fixes #1152 

 Changes:	
- Added start waitgroup to Proc
- Added app-status to vpn-server, skychat, skysocks client and server 
- Updated methods appCtx, getApp and putApp of Hypervisor

How to test this PR:
1. Add the line `time.Sleep(time.Second*10)` before [this](https://github.com/ersonp/skywire/blob/cdbf44d379dd97ac115389df3cd9b1045624a69b/internal/vpn/server.go#L146)
2. Build `make build`
3. Start hypervisor `./skywire-visor -c skywire-config.json`
4. In the UI stop and restart the `skysocks` or `skychat` app and check the status
5. Start the `vpn-server` (indicator should only turn green after 10 seconds and also the entry to https://sd.skywire.dev/api/services?type=vpn should also be added after 10 seconds)
